tests/director/tests/formation_api_test.go

```diff
@@ -563,6 +571,193 @@ func TestTenantFormationFlow(t *testing.T) {
+			t.Logf("Assert scenarios label after assigning tenant with ID: %q to formation", subscriptionProviderSubaccountID)
+			checkRuntimeFormationLabels(t, ctx, providerRuntime.ID, "scenarios", []string{providerFormationName})
+
+			t.Log("Assert kyma runtime has NOT scenarios label")
+			appRequest := fixtures.FixGetRuntimeRequest(providerRuntime.ID)
+			rtm := graphql.RuntimeExt{}
+			err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, appRequest, &rtm)
+			require.NoError(t, err)
+
+			scenariosLabel, hasScenario := rtm.Labels["scenarios"].([]interface{})
+			require.False(t, hasScenario)
+			require.Empty(t, scenariosLabel)
+		})
+	})
+}
+
+func assignTenantToFormation(t *testing.T, ctx context.Context, objectID, tenantID, formationName string) {
+	t.Logf("Assign tenant %s to formation %s...", objectID, formationName)
+	assignReq := fixtures.FixAssignFormationRequest(objectID, "TENANT", formationName)
+	var formation graphql.Formation
+	err := testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tenantID, assignReq, &formation)
+	require.NoError(t, err)
+	require.Equal(t, formationName, formation.Name)
+	t.Logf("Successfully assigned tenant %s to formation %s", objectID, formationName)
+}
+
+func unassignTenantFromFormation(t *testing.T, ctx context.Context, objectID, tenantID, formationName string) {
+	t.Logf("Unassign tenant %s from formation %s...", objectID, formationName)
+	unassignReq := fixtures.FixUnassignFormationRequest(objectID, "TENANT", formationName)
+	var formation graphql.Formation
+	err := testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tenantID, unassignReq, &formation)
+	require.NoError(t, err)
+	require.Equal(t, formationName, formation.Name)
+	t.Logf("Successfully unassigned tenant %s from formation %s", objectID, formationName)
+}
+
+func createFormationTemplate(t *testing.T, ctx context.Context, tenantID, prefix, formationTemplateName, runtimeType string, runtimeArtifactKind graphql.ArtifactType) {
+	formationTmplInput := graphql.FormationTemplateInput{
+		Name:                   formationTemplateName,
+		ApplicationTypes:       []string{prefix + "-app-type-1", prefix + "-app-type-2"},
+		RuntimeType:            runtimeType,
+		RuntimeTypeDisplayName: prefix + "-formation-template-display-name",
+		RuntimeArtifactKind:    runtimeArtifactKind,
+	}
+
+	formationTmplGQLInput, err := testctx.Tc.Graphqlizer.FormationTemplateInputToGQL(formationTmplInput)
+	require.NoError(t, err)
+	formationTmplRequest := fixtures.FixCreateFormationTemplateRequest(formationTmplGQLInput)
+
+	ft := graphql.FormationTemplate{}
+	t.Logf("Creating formation template with name: %q", formationTemplateName)
+	err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tenantID, formationTmplRequest, &ft)
+	require.NoError(t, err)
+	defer fixtures.CleanupFormationTemplate(t, ctx, certSecuredGraphQLClient, ft.ID)
```

Comment:
This needs to be outside otherwise the template will be deleted immediately

Reply:
yep, I found it when I was validating the test locally